### PR TITLE
Cache sslcontext

### PR DESCRIPTION
**Pull request by tokle**

### CODE REVIEW

**Review by bjorncs — CHANGES_REQUESTED** [review]
Extend abstract component and add a `deconstruct()` method to shutdown executor.

File: vespa-athenz/src/main/java/com/yahoo/vespa/athenz/identity/SiaIdentityProvider.java [hunk 1]
```diff
@@ -14,22 +15,36 @@
...
-public class SiaIdentityProvider implements AthenzIdentityProvider {
+public class SiaIdentityProvider extends AbstractComponent implements AthenzIdentityProvider {
+
+    private static final Duration REFRESH_INTERVAL = Duration.ofHours(1);
 
     private final AthenzDomain domain;
     private final AthenzService service;
     private final String path;
     private final String trustStorePath;
+    AtomicReference<SSLContext> sslContext;
```
Comment (mpolden) [inline vespa-athenz/src/main/java/com/yahoo/vespa/athenz/identity/SiaIdentityProvider.java]: `private final`?

**Review by bjorncs — APPROVED** [review]

**Comment by bjorncs** [comment]
👍

File: vespa-athenz/src/main/java/com/yahoo/vespa/athenz/identity/SiaIdentityProvider.java (hunk 1 above)
Comment (bjorncs) [reply vespa-athenz/src/main/java/com/yahoo/vespa/athenz/identity/SiaIdentityProvider.java]: Also initialize it to something.

**Pull request merged**